File: backend/internal/domain/academic.go
package domain

import "time"

// AcademicYear คือปีการศึกษา (พ.ศ.)
type AcademicYear struct {
	ID        string
	SchoolID  string
	Year      int
	IsCurrent bool
	CreatedAt time.Time
	UpdatedAt time.Time
}

// Semester คือภาคเรียน (1/2) ของปีการศึกษา
type Semester struct {
	ID             string
	SchoolID       string
	AcademicYearID string
	Year           int // ปีการศึกษาของเทอมนี้ (join)
	Term           int
	StartDate      *time.Time
	EndDate        *time.Time
	IsCurrent      bool
	CreatedAt      time.Time
	UpdatedAt      time.Time
}

// NewSemester payload สร้างภาคเรียน
type NewSemester struct {
	AcademicYearID string
	Term           int
	StartDate      *time.Time
	EndDate        *time.Time
}

File: frontend/src/shared/schemas/academic.ts
import { z } from "zod";

// ปีการศึกษา — ตรงกับ backend service.AcademicYearDTO
export const academicYearSchema = z.object({
  id: z.string(),
  year: z.number(),
  is_current: z.boolean(),
});
export type AcademicYear = z.infer<typeof academicYearSchema>;
export const academicYearListSchema = z.array(academicYearSchema);

// ภาคเรียน — ตรงกับ backend service.SemesterDTO
export const semesterSchema = z.object({
  id: z.string(),
  academic_year_id: z.string(),
  year: z.number(),
  term: z.number(),
  start_date: z.string(),
  end_date: z.string(),
  is_current: z.boolean(),
});
export type Semester = z.infer<typeof semesterSchema>;
export const semesterListSchema = z.array(semesterSchema);

export type CreateYearBody = { year: number };
export type CreateSemesterBody = {
  academic_year_id: string;
  term: number;
  start_date: string;
  end_date: string;
};

// semesterLabel ป้ายชื่อภาคเรียนแบบไทย
export function semesterLabel(s: { year: number; term: number }): string {
  return `ปีการศึกษา ${s.year} ภาคเรียนที่ ${s.term}`;
}
